### PR TITLE
Feature/iat 3204

### PR DESCRIPTION
**Pull request by jeffjohnson9046**
[IAT-3204](https://smarterbalanced.atlassian.net/browse/IAT-3204):  Machine Scoring enabled for items that do not support it.

* Only **EQ**, **GI** and **TI** should have `machineScoringManagedByIat` set to **true**
* All other item types should have should have `machineScoringManagedByIat` set to **false**

### CODE REVIEW

File: src/test/java/org/opentestsystem/ap/migration/migration/Migration3204Test.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        assertThatThrownBy(() -> migration3204.migrateEntity(giItemEntity, migrationContext))
+                .isInstanceOf(SkipMigration.class);
+
+        // Skip migration for a TI item entity
+        assertThatThrownBy(() -> migration3204.migrateEntity(tiItemEntity, migrationContext))
+                .isInstanceOf(SkipMigration.class);
+    }
+
+    @Test
+    public void shouldSetTheMachineScoringManagedByIatToFalseForAnAppropriateItem() {
```
Comment (jeffjohnson9046) [inline src/test/java/org/opentestsystem/ap/migration/migration/Migration3204Test.java]: It almost seems like I could lump this test in with what's above and have one unit test that says "this migration should work correctly", but I kinda like having the "should migrate this thing" and "should not migrate these things" in separate tests.

File: src/main/java/org/opentestsystem/ap/migration/migration/Migration3204.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * TIMS.
+ * <p>
+ *     Refer to https://smarterbalanced.atlassian.net/wiki/spaces/IAT/pages/6815825/Scoring+Tab+Rules for the details on
+ *     which item types have their machine scoring handled by TIMS.
+ * </p>
+ */
+@Component
+public class Migration3204 extends AbstractMigration {
+    // A list of the item types that have their machine scoring managed by TIMS.
+    private final Collection<String> itemTypesToExclude = Arrays.asList(
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/migration/migration/Migration3204.java]: Add `static`?

File: src/main/java/org/opentestsystem/ap/migration/migration/Migration3204.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                dataStoreUtility,
+                dataStoreAttachmentManager);
+    }
+
+    @Override
+    protected ItemEntity migrateEntity(final ItemEntity itemEntity, final MigrationContext migrationContext) {
+        // if the item is an EQ, GI or TI:  skip migration - items of these types should not be touched by this
+        // migration.
+        if (itemTypesToExclude.contains(itemEntity.getItemJson().getType())) {
+            throw new SkipMigration(String.format("Item id %s is a(n) %s item, thus should not be affected by this migration.",
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/migration/migration/Migration3204.java]: Love the grammar here.  "thus" :)

File: src/main/java/org/opentestsystem/ap/migration/migration/Migration3204.java (hunk 2 above)
Comment (jeffjohnson9046) [reply src/main/java/org/opentestsystem/ap/migration/migration/Migration3204.java]: Indeed - the very definition of "static".  I'll make that update.

File: src/test/java/org/opentestsystem/ap/migration/migration/Migration3204Test.java (hunk 1 above)
Comment (millerts) [reply src/test/java/org/opentestsystem/ap/migration/migration/Migration3204Test.java]: I prefer the way you have it.  IMHO, each test should verify a specific logic path.

**Pull request merged**